documentation/asciidoc/topics/proc_batching_configmap.adoc

```diff
@@ -0,0 +1,54 @@
+[id='batching-configmap_{context}']
+= Running Batch Scripts with Configuration Files
+Use a `ConfigMap` for batch operations if you want to use additional files such as an `infinispan.xml` cache configuration.
+
+.Procedure
+
+. Create a batch script named `batch` that contains all the CLI operations you want to run, for example:
++
+[source,options="nowrap",subs=attributes+]
+----
+create cache --file=mycache.xml mycache
+put --cache=mycache hello world
+put --cache=mycache hola mundo
+disconnect
+----
++
+. Save `batch` and all configuration artifacts that it requires to a local directory that you can use as a `ConfigMap`, for example:
++
+[source,options="nowrap",subs=attributes+]
+----
+/tmp/mybatch
+├── batch
+└── mycache.xml
+----
++
+. Create a `Batch` CR.
+.. Specify the name of the {brandname} cluster where you want to run the batch operations with the `spec.cluster` field.
+.. Set the name of the `ConfigMap` that contains your batch file and configuration artifacts with the `spec.configMap` field.
++
+[source,yaml,options="nowrap",subs=attributes+]
+----
+include::yaml/batch_configmap.yaml[]
+----
++
+. Apply your `Batch` CR.
++
+[source,options="nowrap",subs=attributes+]
+----
+$ {oc_apply_cr} my-batch.yaml
+----
++
+. Create a `ConfigMap` from the local directory that contains the `batch` script, for example:
++
+[source,options="nowrap",subs=attributes+]
+----
+$ {oc_configmap} batch-config-map --from-file=/tmp/mybatch
```

Comment:
Probably, we can also provide whole ConfigMap example as yaml to increase visiblity, not only compiled with `oc/kubect` command

Reply:
@oraNod didn't find whole ConfigMap example in addition to create ConfigMap from the folder

Reply:
@dmvolod Hey, I noticed your initial comment and have spent a little time investigating. Can you explain a little more about what you mean for "whole ConfigMap" example please?

Reply:
Oh, sorry, probably miss something, I talked about example like this
```
apiVersion: v1
kind: ConfigMap
metadata:
  name: batch-config-map
data:
  batch: |
    create cache --file=mycache.xml mycache
    put --cache=mycache hello world
    put --cache=mycache hola mundo
  mycache.xml: |
    <infinispan>
        <cache-container>
            <distributed-cache name="mycache" mode="SYNC">
              <encoding media-type="application/x-protostream"/>
              <memory max-count="1000000" when-full="REMOVE"/>
            </distributed-cache>
        </cache-container>
    </infinispan>
```
Someone can create it from the files folder, someone manually

Reply:
Thanks, @dmvolod. I didn't think about creating the configmap manually. Honestly I think we should discourage that. It's combining yaml and xml, which is not great. 

Cache configuration needs to be valid and it's easier to do that if xml or json files are in standalone files. I've actually added a bit of doc to this PR to address that recommendation. 

@ryanemerson Could you please give your 2c here?

Reply:
+1 to preferring the individual file approach. @dmvolod suggestion is totally valid, and users who are familiar with config maps may choose to use this approach, but I don't think we need to document every possible option just our recommendation.

Reply:
Thanks @ryanemerson . Let's go with that. @dmvolod Cheers again for the review and good suggestion.